src/binary-reader.cc

```diff
@@ -2172,7 +2172,7 @@ Result BinaryReader::ReadElemSection(Offset section_size) {
     ERROR_IF(flags > ~(~0u << SegFlagMax), "invalid elem segment flags: %#x",
              flags);
     Index table_index(0);
-    if (flags & SegExplicitIndex) {
+    if ((flags & (SegPassive | SegExplicitIndex)) == SegExplicitIndex) {
```

Comment:
Is this needed as part of this change?

Should we instead assert inside the if block if SegPassive is set?

Reply:
We'll want to support declared segments soon, so yeah. Maybe should error instead of assert.